wordpress-mobile/WordPress-iOS
# Offline Support: Posts no longer removed when discarding changes.

### Descritpion:

Fixes #9985.

Posts were being removed from the local list if the device was offline and the user discarded changes to the post under certain conditions (steps described in #9985).

### Details:

This PR changes two things:

1. It removes the `shouldRemoveOnDismiss` from `AbstractPost`.  `AbstractPost` should not be aware or define VC logic.
2. Removes a condition for the post removal logic.  I analyzed the condition trying to understand if it's necessary, and couldn't really come up with any reason to keep that logic.

### Testing:

1. Edit an existing published post
2. Turn on airplane mode
3. Change some text
4. Publish & exit the editor
5. You should see the post with the changes and an "Upload failed" message.
6. Open the post again
7. Make some more changes.
8. Try to exit, and discard the changes.
9. The post should be back to what it at the last successful syncrhonization.
